onnx_tf/common/tf_helper.py

```diff
@@ -2,21 +2,22 @@
 import numpy as np
 
 
-def tf_shape(tensor):
-    """
+def tf_shape(tensor, dtype=tf.int64):
+  """
         Helper function returning the shape of a Tensor.
         The function will check for fully defined shape and will return
         numpy array or if the shape is not fully defined will use tf.shape()
         to return the shape as a Tensor.
     """
-    if tensor.shape.is_fully_defined():
-        return np.array(tensor.shape.as_list(), dtype=np.int64)
-    else:
-        return tf.shape(tensor, out_type=tf.int64)
+  if tensor.shape.is_fully_defined():
+    TF_TYPE_TO_NP_TYPE = {tf.int32: np.int32, tf.int64: np.int64}
+    return np.array(tensor.shape.as_list(), dtype=TF_TYPE_TO_NP_TYPE[dtype])
```

Comment:
Not sure the dictionary is needed. Would dtype.as_numpy_dtype work?

Reply:
@chinhuang007 updated according to your comment